spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -115,21 +115,21 @@ module.exports = function (provider, setup) {
       var paused = false;
       var messageCount = 0;
       dispatch.on('onMessage', function (msg) {
-        if (!('data' in msg)) {
+        if (msg && !(msg.hasOwnProperty('data'))) {
           // Not an 'onData' message.
           return;
         }
 
         // One onData is allowed during pause due to https://crbug.com/360026
-        ++messageCount;
-        if (paused && messageCount === 1) {
+        if (paused && ++messageCount === 1) {
```

Comment:
It seems like you're changing the definition of `messageCount`?  I don't understand the advantage, but if you prefer this pattern, then I think you should change the name to something like `messagesReceivedWhilePaused`.